### ISSUE TITLE
Canceled COPY returns with wrong sqlstate

### ISSUE THREAD

**Issue by dvarrazzo** [cockroachdb/cockroach#81559]
**Describe the problem**

If a copy operation is canceled (using [`PQputCopyEnd()`](https://www.postgresql.org/docs/current/libpq-copy.html#LIBPQ-PQPUTCOPYEND) with a message), the error returned has sqlstate `XXUUU` instead of `57104`. This results in psycopg raising an `InternalError` rather than `QueryCanceled`. The error message passed as copy end argument is lost, leaving the user puzzled.

**To Reproduce**

```python
>>> import psycopg
>>> conn = psycopg.connect("postgresql://root@localhost:26257/defaultdb?sslmode=disable")
>>> cur = conn.cursor()
>>> cur.execute("create table whatever (id integer primary key)")
>>> with cur.copy("copy whatever from stdin") as copy:
...     1 / 0

Traceback (most recent call last):
  File "/home/piro/dev/psycopg3/psycopg/psycopg/cursor.py", line 878, in copy
    yield copy
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/usr/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/piro/dev/psycopg3/psycopg/psycopg/cursor.py", line 880, in copy
    raise ex.with_traceback(None)
psycopg.InternalError: client canceled COPY
```

WIth normal postgres instead:

```python
>>> import psycopg
>>> conn = psycopg.connect("")
>>> cur = conn.cursor()
>>> cur.execute("create table whatever (id integer primary key)")
>>> with cur.copy("copy whatever from stdin") as copy:
...     1 / 0
 
Traceback (most recent call last):
  File "/home/piro/dev/psycopg3/psycopg/psycopg/cursor.py", line 878, in copy
    yield copy
  File "<stdin>", line 2, in <module>
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/usr/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/piro/dev/psycopg3/psycopg/psycopg/cursor.py", line 880, in copy
    raise ex.with_traceback(None)
psycopg.errors.QueryCanceled: COPY from stdin failed: error from Python: ZeroDivisionError - division by zero
CONTEXT:  COPY whatever, line 1
```

**Environment:**
 - CockroachDB version 21.2.10
 - Server OS: Docker image
 - Python + psycopg 3

Epic CRDB-14049

Jira issue: CRDB-15078

**Comment by otan** [comment]
will be resolved by #81564

**Comment by dvarrazzo** [comment]
Amazing, thank you!
